bin/add_frontend_server.rb

```diff
@@ -182,8 +182,8 @@ def generate_instance_zone_and_name(ec2client, ssh_username, frontend_name = nil
                                                     monitoring: {
                                                         enabled: true
                                                     },
-                                                    # Prevent api termination, except for adhoc instances.
-                                                    disable_api_termination: (environment != 'adhoc'),
+                                                    # Don't disable api termination.
+                                                    disable_api_termination: false,
```

Comment:
Yes, I will do that in my next PR.  (This one is pretty big and multifaceted already)